### PR TITLE
(maint) update exists? -> exist? for ruby 3.2 compatibility

### PR DESCRIPTION
**Pull request by tvpartytonight**

### CODE REVIEW

File: lib/in_parallel.rb [hunk 1]
```diff
@@ -185,7 +185,7 @@ def self._execute_in_parallel(method_sym, obj = self, &block)
       ret_val                   = nil
       # Communicate the return value of the method or block
       read_result, write_result = IO.pipe
-      Dir.mkdir('tmp') unless Dir.exists? 'tmp'
+      Dir.mkdir('tmp') unless Dir.exist? 'tmp'
```
Comment (ekohl) [inline lib/in_parallel.rb:188]: I read through this and didn't want to make too many changes, but I did think about using `FileUtils.mkdir_p`.

File: lib/in_parallel.rb [hunk 2]
```diff
@@ -152,7 +152,7 @@ def self.wait_for_processes(proxy = self, binding = nil, timeout = nil, kill_all
                 end
                 results_map[process_info[:index]] = { process_info[:tmp_result] => marshalled_result }
               ensure
-                File.delete(process_info[:std_out]) if File.exists?(process_info[:std_out])
+                File.delete(process_info[:std_out]) if File.exist?(process_info[:std_out])
```
Comment (ekohl) [inline lib/in_parallel.rb:155]: The atomic `FileUtils.rm_f` is actually a better way, but I'm not sure how many changes you want to make here.

**Review by joshcooper — APPROVED** [review]
I don't know anything about this repo, but the changes look good to me. Not sure why PR tests failed?

**Comment by tvpartytonight** [comment]
The failure is a due to a pinning rspec to 3.1.0: https://stackoverflow.com/questions/35893584/nomethoderror-undefined-method-last-comment-after-upgrading-to-rake-11 https://jenkins-sre.delivery.puppetlabs.net/job/qe_in-parallel_unit-ruby_ghpr_master/RUBY_VER=ruby-2.5.9,UNEEDED=UNEEDED,label=k8s-unit/3/console

I will loosen the version required so this gem pulls the latest rspec versions.

**Comment by tvpartytonight** [comment]
I had to re-run the 2.7 cell a few times; there was an error with one test but I couldn't reproduce it locally and now it seems fixed in the gh action. 🤷

**Comment by ekohl** [comment]
@tvpartytonight @joshcooper any chance you could make this into a release? It blocks Beaker running on Ruby 3.2 (https://github.com/voxpupuli/beaker/pull/1762).

**Pull request merged**